WebCryptoAPI/derive_bits_keys/cfrg_curves_bits.js

```diff
@@ -0,0 +1,245 @@
+              return subtle.deriveBits({name: algorithmName.toLowerCase(), public: publicKeys[algorithmName]}, privateKeys[algorithmName], 8 * sizes[algorithmName])
+              .then(function(derivation) {
+                  assert_true(equalBuffers(derivation, derivations[algorithmName]), "Derived correct bits");
+              }, function(err) {
+                  assert_unreached("deriveBits failed with error " + err.name + ": " + err.message);
+              });
+          }, algorithmName + " mixed case parameters");
+
+          // Null length
+          promise_test(function(test) {
+              return subtle.deriveBits({name: algorithmName, public: publicKeys[algorithmName]}, privateKeys[algorithmName], null)
+              .then(function(derivation) {
+                  assert_true(equalBuffers(derivation, derivations[algorithmName]), "Derived correct bits");
+              }, function(err) {
+                  assert_unreached("deriveBits failed with error " + err.name + ": " + err.message);
+              });
+          }, algorithmName + " with null length");
+
+          // Shorter than entire derivation per algorithm
+          promise_test(function(test) {
+              return subtle.deriveBits({name: algorithmName, public: publicKeys[algorithmName]}, privateKeys[algorithmName], 8 * sizes[algorithmName] - 32)
+              .then(function(derivation) {
+                  assert_true(equalBuffers(derivation, derivations[algorithmName], 8 * sizes[algorithmName] - 32), "Derived correct bits");
+              }, function(err) {
+                  assert_unreached("deriveBits failed with error " + err.name + ": " + err.message);
+              });
+          }, algorithmName + " short result");
+
+          // Non-multiple of 8
+          promise_test(function(test) {
+              return subtle.deriveBits({name: algorithmName, public: publicKeys[algorithmName]}, privateKeys[algorithmName], 8 * sizes[algorithmName] - 11)
+              .then(function(derivation) {
+                  assert_true(equalBuffers(derivation, derivations[algorithmName], 8 * sizes[algorithmName] - 11), "Derived correct bits");
+              }, function(err) {
+                  assert_unreached("deriveBits failed with error " + err.name + ": " + err.message);
+              });
+          }, algorithmName + " non-multiple of 8 bits");
+
+          // Errors to test:
+
+          // - missing public property TypeError
+          promise_test(function(test) {
+              return subtle.deriveBits({name: algorithmName}, privateKeys[algorithmName], 8 * sizes[algorithmName])
+              .then(function(derivation) {
+                  assert_unreached("deriveBits succeeded but should have failed with TypeError");
+              }, function(err) {
+                  assert_equals(err.name, "TypeError", "Should throw correct error, not " + err.name + ": " + err.message);
+              });
+          }, algorithmName + " missing public curve");
```

Comment:
I know the original file has the same typo but I guess this should be
```suggestion
          }, algorithmName + " missing public property");
```